synapse/storage/events.py

```diff
@@ -539,16 +553,20 @@ def _get_new_state_after_events(self, room_id, events_context, old_latest_event_
                 the new forward extremities for the room.
 
         Returns:
-            Deferred[dict[(str,str), str]|None]:
-                None if there are no changes to the room state, or
-                a dict of (type, state_key) -> event_id].
+            Deferred[tuple[dict[(str,str), str]|None, dict[(str,str), str]|None]]:
+            Returns a tuple of two state maps, the first being the full new current
+            state and the second being the delta to the existing current state.
+            If both are None then there has been no change.
```

Comment:
can we clarify when the second is None but the first is not?

Reply:
(done)